common/lockmanager_test.go

```diff
@@ -0,0 +1,114 @@
+package common
+
+import (
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/edulinq/autograder/config"
+)
+
+var (
+    key = "testkey1"
+    key2 = "testkey2"
+    doesNotExistKey = "dne"
+    staleDuration = time.Duration(config.STALELOCK_DURATION.Get()) * time.Second;
+    wg sync.WaitGroup;
+)
+
+func TestLockManager(test *testing.T) {
+    testLockingKey()
+    
+    testUnlockingKey(test)
+
+    testConcurrentLockingUnlocking()
+
+    testUnlockingAnUnlockedLock(test)
+    
+    testUnlockingMissingKey(test)
+
+    testLockConcurrencyWithStaleCheck(test)
+}
+
+func testLockingKey() {
+    Lock(key)
```

Comment:
This test cannot fail.
There is not much point in a test that cannot fail.

Since our lock manager should never panic (which is correct for most go code),
you will always need to call some method from *testing.T.

See comments on `testUnlockingKey()`.